### ISSUE TITLE
Issues with transparent background and kitty after new update

### ISSUE THREAD

**Issue by manangulati9** [catppuccin/nvim#791]
### Description

I was using kitty opacity 0.75 and nvim didn't follow this opacity which was fine for me. After #786 if I want to revert to the old behavior, the option for that seems to be missing. I tried using `transparent_background=false` but that only reverts colors for plugins and not the main editor itself. Also, I'd appreciate some help on how to set LSP hover info box border with `transparent_background` enabled as it's not present atm. The black color for LSP info is also not ideal. I'd like to shift to this new transparent theme but the two issues above preventing me for doing so right now.

`transparent_background` enabled
![241027_18h11m04s_screenshot](https://github.com/user-attachments/assets/3df2916c-8eab-4fed-a03b-7357b12c843b)
![241027_18h11m35s_screenshot](https://github.com/user-attachments/assets/8c052a28-24e9-4db6-9f13-1b5d6a0e4aae)

`transparent_background` disabled
![241027_18h12m27s_screenshot](https://github.com/user-attachments/assets/b677e9b5-ef82-4922-b1c9-f25fe5d59daf)

### Neovim version

```markdown
NVIM v0.10.2
Build type: RelWithDebInfo
LuaJIT 2.1.1727870382
```

### Terminal and multiplexer

kitty 0.36.4 with tmux 3.5a

### Catppuccin version / branch / rev

cf021eb

### Steps to reproduce

nvim -u repro.lua

### Expected behavior

Setting `transparent_background = false` should prevent catppuccin from following kitty's opacity

### Actual behavior

Setting `transparent_background = false` only resets colors for plugins, not the main editor itself.

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "catppuccin/nvim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("catppuccin")
-- add anything else here
```

**Comment by manangulati9** [comment]
As suggested in the discord server, I also tried setting `kitty = false` but that also doesn't do anything. Also, I encountered a weird bug when opening oil.nvim during the landing page. Somehow there's some new weird lines appearing. It maybe an issue with oil.nvim I'm not entirely sure.
![241027_19h10m19s_screenshot](https://github.com/user-attachments/assets/2b240bda-ad9f-4a92-9389-f2005e85f514)
[Kooha-2024-10-27-19-16-40.webm](https://github.com/user-attachments/assets/f23f922f-ed6e-4146-9dde-1a412cd7daae)

**Comment by sgoudham** [comment]
Hey :wave:,

If I'm understanding this correctly, we merged #786 to only adjust the colour channels (and therefore workaround the kitty issue documented in the source code) if `kitty` is true and if `transparent_background` is true.

Could you try setting both those to true in your configuration and see what results you get? 

Please let me know if I'm still misunderstanding your issue and what you want to achieve.
